### PR TITLE
[ci] Add rspec test for project watchlist

### PR DESCRIPTION
**Pull request by bgeuken**
* Migrate project watchlist tests to rspec test suite
* Fix caching issue that appears when changing watched projects within a second,
  first found in 82e34072cd0e6dd2600e5360...

### CODE REVIEW

File: src/api/spec/features/webui/watchlists_spec.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
+require "browser_helper"
+
+RSpec.feature "Watchlists", :type => :feature, :js => true do
+  let!(:user) { create(:confirmed_user, login: "kody") }
```
Comment (ChrisBr) [inline src/api/spec/features/webui/watchlists_spec.rb]: Do we really need the exclamation mark?

Comment (bgeuken) [inline src/api/spec/features/webui/watchlists_spec.rb]: We don't. I'll remove it

File: src/api/spec/factories/users.rb [hunk 2]
```diff
@@ -11,6 +11,14 @@
       factory :admin_user do
         roles { [Role.find_by_title('admin')] }
       end
+
+      factory :user_with_watched_project do
```
Comment (hennevogel) [inline src/api/spec/factories/users.rb]: How often do you think this will happen?

File: src/api/spec/features/webui/watchlists_spec.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  end
+
+  scenario "remove projects from watchlist" do
+    login user_with_watched_project
+    visit project_show_path(project: "brian_s_watched_project")
+
+    page.execute_script("$('#menu-favorites').show();")
+    expect(page).to have_content("List of projects you are watching")
+    expect(page).to have_css("a span.project-link", text: "brian_s_watched_project")
+    expect(page).to have_css("#menu-favorites", text: "Remove this project from Watchlist")
```
Comment (hennevogel) [inline src/api/spec/features/webui/watchlists_spec.rb]: What's the value of testing also for the texts?

File: src/api/spec/features/webui/watchlists_spec.rb [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+    find(:css, "#toggle_watch > span.desc").click
+
+    visit project_show_path(project: "brian_s_watched_project")
+    page.execute_script("$('#menu-favorites').show();")
+    expect(page).to have_no_css("a span.project-link")
+    expect(page).to have_css("#menu-favorites", text: "Add this project to Watchlist")
```
Comment (hennevogel) [inline src/api/spec/features/webui/watchlists_spec.rb]: Same here, why test for the text?

File: src/api/spec/features/webui/watchlists_spec.rb (hunk 3 above)
Comment (bgeuken) [inline src/api/spec/features/webui/watchlists_spec.rb]: What's the value of testing views with capybara?

Comment (bgeuken) [inline src/api/spec/features/webui/watchlists_spec.rb]: I'll move it to the spec

File: src/api/spec/factories/users.rb (hunk 2 above)
Comment (bgeuken) [inline src/api/spec/factories/users.rb]: I'll move it to the spec

**Comment by mdeniz** [comment]
LGTM :+1:

**Comment by mdeniz** [comment]
@hennevogel review it and merge it if is ok for you!

**Comment by ChrisBr** [comment]
LGTM too!

**Pull request merged**